src/middleware/AppContext.tsx

```diff
@@ -32,6 +36,8 @@ export const ContextProvider = ({ children }: { children: ReactNode }) => {
   const chrome = useChrome();
   const [contentOrigin, setContentOrigin] = useState<ContentOrigin>(ContentOrigin.CUSTOM);
   const { fetchFeatures, isLoading: isFetchingFeatures } = useFetchFeaturesQuery();
+  const [subscriptions, setSubscriptions] = useState<Subscriptions | null>(null);
+  const { fetchSubscriptions, isLoading: isFetchingSubscriptions } = useFetchSubscriptionsQuery();
```

Comment:
By using react query we can just use their in built hooks to take care of the state, so we won't need to bloat the state/useEffect here at all.